enclone_args/src/read_json.rs

```diff
@@ -175,126 +133,89 @@ fn parse_vector_entry_from_json(
     let mut cdr3_aa: String;
     let mut cdr3_dna: String;
     let mut cdr3_start: usize;
-    if v.get("version").is_some() {
-        *cr_version = v["version"].to_string().between("\"", "\"").to_string();
-    }
 
-    // Read validated and non-validated UMIs.
-
-    let mut validated_umis = Vec::<String>::new();
-    let mut validated_umis_present = false;
-    let val = v["validated_umis"].as_array();
-    if let Some(val) = val {
-        validated_umis_present = true;
-        for vi in val {
-            validated_umis.push(vi.to_string().between("\"", "\"").to_string());
-        }
-    }
-    let mut non_validated_umis = Vec::<String>::new();
-    let mut non_validated_umis_present = false;
-    let non_val = v["non_validated_umis"].as_array();
-    if let Some(non_val) = non_val {
-        non_validated_umis_present = true;
-        for nv in non_val {
-            non_validated_umis.push(nv.to_string().between("\"", "\"").to_string());
-        }
-    }
-    let mut invalidated_umis = Vec::<String>::new();
-    let mut invalidated_umis_present = false;
-    let inval = v["invalidated_umis"].as_array();
-    if let Some(inval) = inval {
-        invalidated_umis_present = true;
-        for inv in inval {
-            invalidated_umis.push(inv.to_string().between("\"", "\"").to_string());
-        }
-    }
-
-    // Read fraction_of_reads_for_this_barcode_provided_as_input_to_assembly.
-
-    let mut frac_reads_used = None;
-    let f = v["fraction_of_reads_for_this_barcode_provided_as_input_to_assembly"].as_f64();
-    if let Some(f) = f {
-        frac_reads_used = Some((f * 1_000_000.0).round() as u32);
-    }
+    let frac_reads_used = ann
+        .fraction_of_reads_for_this_barcode_provided_as_input_to_assembly
+        .map(|f| (f * 1_000_000.0).round() as u32);
```

Comment:
Wonder what is happening here with the 1M.

Reply:
🤷